libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -317,6 +317,12 @@ uint8_t AC_Fence::check()
         return 0;
     }
 
+    //check for non-enabled fence breach. Can happen if fence type is changed in flight. 
+    if (_breached_fences) {
+        if (!(_breached_fences & _enabled_fences)) {
+            clear_breach(~_enabled_fences);
```

Comment:
Couldn't we just skip to this bit?  It's not expensive.

Reply:
Its a lot neater this way. I have updated the PR. Thanks :)